app/boards/shields/settings_reset/settings_reset.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2023 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+
+#include <zmk/settings.h>
+
+static int reset_settings_init(const struct device *dev) {
```

Comment:
Done. (I originally didn't do this because you could just add this shield to any build to get its special behavior, but I don't really have a strong opinion either way, and having the new config be able to `select SETTINGS` does seem a bit cleaner.)